### PR TITLE
feat: replace Recharts with visx+motion animated AreaChart

### PR DESCRIPTION
**Pull request by Sensei3k**
## Summary

- Replaces the static Recharts bar/line charts in \`CyclePerformanceChart\` with a spring-animated dual-area chart built on \`@visx\` + \`motion/react\`
- Extracts \`buildChartData\` into \`lib/chart-data.ts\` with 13 unit tests (kobo→NGN, date parsing, cumulative running total) — full RED→GREEN TDD cycle
- Adds \`components/ui/area-chart.tsx\` — the full visx+motion primitive (crosshair, animated tooltip, date ticker pill, per-area gradient fills)
- Adds semantic \`--chart-*\` CSS tokens in \`globals.css\` mapped to existing ajo design system vars (\`--ajo-paid\`, \`--ajo-outstanding\`)

## What changed

| File | Change |
|------|--------|
| \`lib/chart-data.ts\` | New — extracted + typed \`buildChartData\` |
| \`tests/unit/buildChartData.test.ts\` | New — 13 unit tests (vitest) |
| \`vitest.config.ts\` | New — scopes vitest to \`tests/unit/\` only |
| \`components/ui/area-chart.tsx\` | New — full visx+motion chart primitive |
| \`components/dashboard/cycle-performance-chart.tsx\` | Rewritten — Recharts removed, AreaChart wired up |
| \`app/globals.css\` | Added \`--chart-*\` semantic tokens |
| \`package.json\` | Added \`@visx/*\`, \`d3-array\`, \`motion\`, \`react-use-measure\`; retained \`recharts@2.15.4\` for \`components/ui/chart.tsx\` (shadcn primitive) |

## Fixes applied post-review

**Data correctness**
- \`new Date('YYYY-MM-DD')\` parses as UTC and shifts the displayed day for users west of UTC — replaced with \`new Date(y, m-1, d)\` (local midnight construction)
- Running cumulative total now accumulates in kobo integers to prevent floating-point drift across many cycles; divides once per datum at return time

**Type safety**
- SVG gradient \`stop-color\` doesn't resolve CSS custom property chains (\`var → var → oklch\`) — brand hex values (\`#00bc7d\`, \`#e8970a\`) passed directly to \`Area fill\` props; HTML elements (legend, tooltip) keep CSS vars which resolve fine
- Tooltip value coercions guarded with \`typeof\` checks + \`0\` fallback to prevent \`₦NaN\` rendering
- Explanatory comment added on the \`as unknown as\` cast

**Accessibility**
- Removed \`sr-only <p>\` elements that duplicated \`aria-label\` on the \`role="img"\` container — screen readers were announcing the label twice

**Axis**
- \`numTicks\` capped at \`Math.min(data.length, 6)\` to prevent label crowding on small viewports or groups with many cycles

**Tests**
- Date test documents local-time intent explicitly
- Added \`cumulative === 0\` assertion for the no-payments edge case

## Test plan

- [x] \`yarn test:unit\` — 13/13 passing
- [x] \`yarn build\` — clean TypeScript compile
- [ ] Per Cycle view: teal (collected) + gold (outstanding) dual area, light mode
- [ ] Per Cycle view: teal + gold on dark surface, dark mode
- [ ] Cumulative view: teal single area, light + dark mode
- [ ] Hover crosshair + animated tooltip with NGN values
- [ ] Animated date ticker pill at bottom on hover
- [ ] View toggle (Per Cycle / Cumulative) switches correctly

### CODE REVIEW

File: tests/dashboard.spec.ts [hunk 1]
```diff
@@ -117,42 +117,53 @@ test.describe('Circle Dashboard', () => {
...
+    test('tooltip appears with ₦ values on hover in Cumulative view', async ({ page }) => {
       const dashboard = new DashboardPage(page);
       await dashboard.goto();
       await dashboard.switchToChartView();
       await dashboard.switchToCumulative();
 
-      // Hover over the SVG area to trigger the line chart tooltip
+      // Wait for the chart animation (1100ms) to complete before interaction is enabled
+      await page.waitForTimeout(1500);
+
```
[the remark this answers is not in the document]
Comment (Sensei3k) [reply tests/dashboard.spec.ts]: Fixed — replaced `waitForTimeout(1500)` with `expect(chartSvg).toHaveAttribute('data-interactive', 'true', { timeout: 5000 })`. The chart SVG now sets `data-interactive="true"` once `isLoaded` flips to true after the animation completes, giving the test a deterministic signal to wait on.

File: components/ui/area-chart.tsx [hunk 2]
```diff
@@ -0,0 +1,2274 @@
+"use client";
+
+import { localPoint } from "@visx/event";
+import { curveMonotoneX } from "@visx/curve";
+import { GridColumns, GridRows } from "@visx/grid";
+import { ParentSize } from "@visx/responsive";
+import { scaleLinear, scaleTime, type scaleBand } from "@visx/scale";
```
[the remark this answers is not in the document]
Comment (Sensei3k) [reply components/ui/area-chart.tsx:7]: This is actually valid TypeScript — `typeof scaleBand` inside `ReturnType<typeof ...>` is a type-level `typeof` query, not a runtime value reference. The `type` modifier on the import only strips the emit; TypeScript resolves `typeof` in type positions at compile time without needing the runtime binding. `tsc --noEmit` confirms no errors, so no change is needed here.

File: components/ui/area-chart.tsx [hunk 3]
```diff
@@ -0,0 +1,2274 @@
...
+  const bisectDate = useMemo(
+    () => bisector<Record<string, unknown>, Date>((d) => xAccessor(d)).left,
+    [xAccessor]
+  );
+
+  const xScale = useMemo(() => {
+    const dates = data.map((d) => xAccessor(d));
+    const minTime = Math.min(...dates.map((d) => d.getTime()));
+    const maxTime = Math.max(...dates.map((d) => d.getTime()));
+
```
[the remark this answers is not in the document]
Comment (Sensei3k) [reply components/ui/area-chart.tsx:2109]: Fixed — added an early return in the `xScale` `useMemo` when `data.length === 0`, providing a safe `[0, 1]` domain instead of computing `Math.min/max` over an empty array which yields `Infinity/-Infinity`.

File: tests/dashboard.spec.ts [hunk 4]
```diff
@@ -117,42 +117,53 @@ test.describe('Circle Dashboard', () => {
...
       await dashboard.switchToChartView();
 
-      // Hover the first bar in the per-cycle chart
-      const firstBar = dashboard.perCycleChartContainer.locator('svg .recharts-bar-rectangle').first();
-      await firstBar.hover();
+      // Scroll chart into view and wait for the 1100ms animation before interaction is enabled
+      const chartSvg = dashboard.perCycleChartContainer.locator('svg').first();
+      await chartSvg.scrollIntoViewIfNeeded();
+      await page.waitForTimeout(1500);
+
```
[the remark this answers is not in the document]
Comment (Sensei3k) [reply tests/dashboard.spec.ts]: Fixed — same as the other tooltip test: replaced `waitForTimeout(1500)` with a `toHaveAttribute('data-interactive', 'true')` assertion on the chart SVG, which is set deterministically once `isLoaded` is true.

File: components/ui/area-chart.tsx [hunk 5]
```diff
@@ -0,0 +1,2317 @@
...
+type ScaleLinearType<Output, _Input = number> = ReturnType<
+  typeof scaleLinear<Output>
+>;
+type ScaleTimeType<Output, _Input = Date | number> = ReturnType<
+  typeof scaleTime<Output>
+>;
+type ScaleBandType<Domain extends { toString(): string }> = ReturnType<
+  typeof scaleBand<Domain>
+>;
```
[the remark this answers is not in the document]
Comment (Sensei3k) [reply components/ui/area-chart.tsx:58]: No change needed — `typeof scaleBand` inside `ReturnType<>` is a compile-time type query, not a runtime value access. TypeScript handles this correctly with type-only imports and `tsc --noEmit` produces zero errors, so the build is not broken.

File: components/ui/area-chart.tsx [hunk 6]
```diff
@@ -0,0 +1,2317 @@
...
+
+  const xScale = useMemo(() => {
+    const dates = data.map((d) => xAccessor(d));
+    const minTime = Math.min(...dates.map((d) => d.getTime()));
+    const maxTime = Math.max(...dates.map((d) => d.getTime()));
+
+    return scaleTime({
+      range: [0, innerWidth],
+      domain: [minTime, maxTime],
+    });
```
[the remark this answers is not in the document]
Comment (Sensei3k) [reply components/ui/area-chart.tsx:2113]: Fixed — added a `data.length === 0` guard at the top of the `xScale` `useMemo` that returns a safe `[0, 1]` domain, preventing `Infinity/-Infinity` from being passed to `scaleTime`.

File: components/dashboard/cycle-performance-chart.tsx [hunk 7]
```diff
@@ -1,98 +1,37 @@
...
 
 export function CyclePerformanceChart({ cycles, payments }: CyclePerformanceChartProps) {
   const [view, setView] = useState<ChartView>('per-cycle');
-  const data = useMemo(() => buildChartData(cycles, payments), [cycles, payments]);
-  const totalCollected = data[data.length - 1]?.cumulative ?? 0;
+  const rawData = useMemo(() => buildChartData(cycles, payments), [cycles, payments]);
+  // AreaChart primitive requires Record<string, unknown>[] — CycleChartDatum is
+  // structurally compatible but lacks an index signature, so a double cast is needed.
+  const data = rawData as unknown as Record<string, unknown>[];
+  const totalCollected = rawData[rawData.length - 1]?.cumulative ?? 0;
```
[the remark this answers is not in the document]
Comment (Sensei3k) [reply components/dashboard/cycle-performance-chart.tsx:35]: Fixed — added an early return in `CyclePerformanceChart` when `rawData.length === 0` that renders a simple empty-state message instead of passing an empty array to `AreaChart`.

File: tests/dashboard.spec.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Sensei3k) [reply tests/dashboard.spec.ts]: Fixed — replaced both `waitForTimeout(1500)` calls in the tooltip tests with `toHaveAttribute('data-interactive', 'true', { timeout: 5000 })`. The chart SVG now exposes `data-interactive="true"` once the post-animation `isLoaded` flag is set, making the wait condition observable and deterministic.

File: tests/dashboard.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sensei3k) [reply tests/dashboard.spec.ts]: Fixed — replaced `waitForTimeout(1500)` with a `toHaveAttribute('data-interactive', 'true')` wait on the cumulative chart SVG, which the chart sets once the animation timer resolves.
